### PR TITLE
Convert Monad.hs to ghc-lib

### PR DESCRIPTION
**Pull request by shayne-fletcher-da**
Convert `Monad.hs` to ghc-lib.

### CODE REVIEW

File: src/GHC/Util/HsExpr.hs [hunk 1]
```diff
@@ -63,17 +64,27 @@ paren' x
...
+isDol' x = isTag' x "$"
+isReturn' x = isTag' x "return" || isTag' x "pure" -- Allow both 'pure' and 'return' as they have the same semantics.
 isRecConstr' (LL _ RecordCon{}) = True; isRecConstr' _ = False
 isRecUpdate' (LL _ RecordUpd{}) = True; isRecUpdate' _ = False
 isDotApp' (LL _ (OpApp _ _ op _)) = isDot' op; isDotApp' _ = False
 isLexeme' (LL _ HsVar{}) = True;isLexeme' (LL _ HsOverLit{}) = True;isLexeme' (LL _ HsLit{}) = True;isLexeme' _ = False
--- Allow both 'pure' and 'return' as they have the same semantics.
-isReturn' (LL _ (HsVar _ (LL _ (Unqual x)))) = occNameString x == "return" || occNameString x == "pure"; isReturn' _ = False
+
+--
```
Comment (googleson78) [inline src/GHC/Util/HsExpr.hs]: Comment with nothing in it here?

**Comment by googleson78** [comment]
Very minor thing but if we're following Haddock markup shouldn't all the code samples be in `@@`s, not in single quotes?

File: src/GHC/Util/HsExpr.hs (hunk 1 above)
Comment (ndmitchell) [reply src/GHC/Util/HsExpr.hs]: I tend to use double newlines rather than `--` breaks

**Comment by ndmitchell** [comment]
Nice!

**Pull request merged**